src/Supportive/Config/CollectorConfig.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\Supportive\Config;
+
+use Qossmic\Deptrac\Core\Layer\Collector\CollectorType;
+
+final class CollectorConfig
+{
+    private ?string $value = null;
+    private bool $private = false;
+    /** @var array<CollectorConfig> */
+    private array $mustNot = [];
+    /** @var array<CollectorConfig> */
+    private array $must = [];
+
+    private function __construct(
+        private readonly CollectorType $collectorType
+    ) {
+    }
+
+    public static function fromType(CollectorType $collectorType): self
+    {
+        $new = new self($collectorType);
+
+        return $new;
+    }
+
+    public function value(string $value): self
+    {
+        $this->value = $value;
+
+        return $this;
+    }
+
+    public function mustNot(CollectorType $collectorType): CollectorConfig
+    {
+        return $this->mustNot[] = CollectorConfig::fromType($collectorType);
+    }
+
+    public function must(CollectorType $collectorType): CollectorConfig
+    {
+        return $this->must[] = CollectorConfig::fromType($collectorType);
+    }
```

Comment:
I am not sure if there is a nicer way to handle this, but it bothers me a bit that all other fluent methods return self, but this one returns the nested CollectorConfig. I imagine this will cause some people to accidentally misconfigure their collectors and I am not sure how easy it is to debug where things went wrong.

Reply:
I think the alternative implementation doesn't suffer from this problem.